src/jit/flowgraph.cpp

```diff
@@ -23084,16 +23099,91 @@ GenTreePtr Compiler::fgInlinePrependStatements(InlineInfo* inlineInfo)
                     }
                     else
                     {
-                        newStmt = gtNewStmt(gtUnusedValNode(argNode), callILOffset);
+                        // In some special cases, unused args with side effects can
+                        // trigger further changes.
+                        //
+                        // (1) If the arg is a static field access, and the field access
+                        // was produced by a call to EqualityComparer<T>.get_Default the
+                        // helper call to ensure the field has a value can be suppressed.
+                        // This helper call is marked as a "Special DCE" helper during
+                        // importation, over in fgGetStaticsCCtorHelper.
+                        //
+                        // (2) NYI. If after, tunneling through GT_RET_VALs, we find that
+                        // the actual arg expression has no side effects, we can skip
+                        // appending all together. This will help jit TP a bit.
+                        //
+                        // Chase through any GT_RET_EXPRs to find the actual argument
+                        // expression.
+                        GenTree* actualArgNode = argNode;
+                        while (actualArgNode->gtOper == GT_RET_EXPR)
+                        {
+                            actualArgNode = actualArgNode->gtRetExpr.gtInlineCandidate;
+                        }
+
+                        // For case (1)
+                        //
+                        // Look for the following tree shapes
+                        // prejit: (IND (ADD (CONST, CALL(special dce helper...))))
+                        // jit   : (COMMA (CALL(special dce helper...), (FIELD ...)))
```

Comment:
Unfortunate that these have different patterns, and jit-diff only shows prejit diffs.  Any thoughts about how to add regression testing for the jit case?

Reply:
We could possibly catch regressions in a perf test. Or SPMI if we ever get that into CI.